### PR TITLE
refs #25285 - support multiple repos in Kickstart

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

**Review by ekohl — APPROVED** [review]
Looks correct to me.

**Comment by lzap** [comment]
I have some comments in core, but have you tried snapshot tests in core? I believe they can fail since `@repo` was unused. Just checking.

**Comment by timogoebel** [comment]
Did you leave out the `--install` parameter for the repo by choice?

**Comment by stbenjam** [comment]
Mostly just thinking about Katello, where we would provide the content through subscription-manager.  I can make `--install` conditional through the additional_media call.

File: provisioning_templates/provision/kickstart_default.erb [hunk 1]
```diff
@@ -61,6 +61,9 @@ if host_param('kickstart_liveimg')
 liveimg --url=<%= liveimg_url %> <%= proxy_string %>
 <% else %>
 <%= @mediapath %><%= proxy_string %>
+% @additional_media.each do |medium| -%>
```
Comment (lzap) [inline provisioning_templates/provision/kickstart_default.erb]:
```suggestion
<% @additional_media.each do |medium| -%>
```

File: provisioning_templates/provision/kickstart_default.erb [hunk 2]
```diff
@@ -61,6 +61,9 @@ if host_param('kickstart_liveimg')
...
+repo --name <%= medium[:name] %> --baseurl <%= medium[:url %> <%= medium[:install] ? ' --install' : '' %>
```
Comment (lzap) [inline provisioning_templates/provision/kickstart_default.erb]:
```suggestion
repo --name <%= medium[:name] %> --baseurl <%= medium[:url] %> <%= medium[:install] ? ' --install' : '' %>
```

**Review by lzap — CHANGES_REQUESTED** [review]
Parsing errors.

But during my testing `@additional_media` is `nil`. Need to dig it up.

**Comment by lzap** [comment]
One important change required, do not use symbols to access keys of repo. Change here or make sure these are hashes with indifferent access.

**Comment by stbenjam** [comment]
I think it's better if the Foreman side makes it indifferent access, we can't be entirely sure what a plugin medium provider might do.  

Apologies this PR wasn't in a good state - should be OK now, but going to reverify with new changes to Foreman side and fix the safe mode issue you noticed.

**Review by lzap — APPROVED** [review]
Sure. NP don't apologize, that's why we do reviews. Looks good.

**Review by lzap — APPROVED** [review]
And it works great! Many thanks, excellent improvement.

**Pull request merged**